### ISSUE TITLE
Kill Switch is Broken

### ISSUE THREAD

**Issue by FlyingWaffleDev** [ProtonVPN/linux-cli#17]
Kill switch does not work.  Or rather, it works _too_ well.  When turning the kill switch on, it breaks the VPN connection and does not allow a reconnection without turning it off.  When attempting the same thing in the GUI, it crashes the application entirely, then on restart shows that kill switch is enabled but also cannot (re)connect.

This is on Gentoo Linux, kernel 5.10.46, protonvpn-cli 3.7.1, gui 1.0.1, and nm-lib 3.3.1.

![Screenshot_2021 06 25_11 37 30](https://user-images.githubusercontent.com/51219918/123457957-0181d100-d5d4-11eb-8c4e-1f3e0d6aa56b.png)

**Comment by calexandru2018** [comment]
Hey @FlyingWaffleDev 

This should not be happening. Could you please contact support our customer support specialists at: https://protonvpn.com/support-form

**Comment by FlyingWaffleDev** [comment]
I put in a support request; went through the process, emailed back and forth, sent log files, did some tests as requested, etc.  In the end support couldn't help, and said they were kicking the issue up to the devs.

Since I'm on Gentoo I'm essentially installing everything myself, so if there's anything else I can try or information I can provide I would be happy to.

**Comment by FlyingWaffleDev** [comment]
So as it turns out, protonvpn relies on a kernel feature that I didn't have enabled.  I had some free time, and managed to track the issue to the "DUMMY" kernel option for 'Dummy net driver support'.  If a kernel is built without this enabled, then protonvpn cannot create the dummy network adapters used for the kill switch feature.

**Comment by calexandru2018** [comment]
@FlyingWaffleDev interesting, that is indeed s plausible explanation. We're you getting the 1000 permission denied error ?

**Comment by FlyingWaffleDev** [comment]
No, the only errors I was seeing were in journalctl or systemctl specifically from NetworkManager when I turned on the Killswitch and reconnected.  I unfortunately didn't record all the errors, but it was something along the lines of "platform-linux do-add-link[ipv6leakintrf0/dummy]: failure 95".

After I rebuilt my kernel it also became apparent that two of protonvpn's network interfaces had not been succesfully created prior to that.  The pvpnksintrf0 and ipv6leakintrf0 interfaces were not shown by ifconfig until I rebuilt my kernel.
